### PR TITLE
Adding soft-delete support for KeyVault certificates

### PR DESCRIPTION
**Pull request by dragav**
This change extends soft-delete support to KeyVault certificates; after this change, and on suitably enabled vaults, deletion of certificates becomes an undoable operation.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@dragav can we get a sign off from someone on your team.

**Comment by dragav** [comment]
@shahabhijeet I have forwarded you emails with signoffs of the team-internal code review. 

The swagger updates were committed in this check-in: https://github.com/dragav/azure-rest-api-specs/commit/09389ad926a9f76cd35a95fbea75474907c1833c. These changes are being withheld currently until we're ready to go public. Thanks.

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Tests/Microsoft.Azure.KeyVault.Tests.csproj [hunk 1]
```diff
@@ -24,13 +24,6 @@
...
   </ItemGroup>
 
-  <ItemGroup>
-    <!--<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-preview-20170106-08" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0-beta5-build1225" />-->
-    <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.2" />
-    <!--<PackageReference Include="xunit" Version="2.2.0-beta5-build3474" />-->
-  </ItemGroup>
-
   <!--<PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">
```
Comment (shahabhijeet) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Tests/Microsoft.Azure.KeyVault.Tests.csproj]: @dragav please remove commented code from project files, they were there for referencing what the project use to target earlier prior to migration. Once you have everything running, feel freed to clean up all the code that might cause confusion in the project files.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by wijays** [comment]
In the latest rev I have started getting this error when trying to get a key from my KeyVault:

TypeLoadException: Could not load type 'Microsoft.Rest.Serialization.UnixTimeJsonConverter' from assembly 'Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

In tracking down the issue I came across this change.

After 'AcquireTokenAsync' completes, when the method (c#) returns to the caller thread, I hit the exception listed above.  Which I believe is related to null property handling in json using converter attributes.

**Comment by dragav** [comment]
Is this still an issue? A TypeLoadException seems to indicate the defining assembly is not present/loaded. The UnixTimeJsonConverter is defined here: SdkCommon\ClientRuntime\ClientRuntime\Serialization\UnixTimeJsonConverter.cs and has been used by KeyVault model entities before the introduction of soft delete; see SDKs\KeyVault\dataPlane\Microsoft.Azure.KeyVault\Generated\Models\IssuerAttributes.cs, for instance.

Given that all tests pass and that this (getting a key) is a fundamental call, I'm inclined to suspect you may have an incomplete SDK installation, or an otherwise non-standard set of binaries.

**Pull request merged**